pkg/controllers/vdb/podfacts.go

```diff
@@ -1086,11 +1086,14 @@ func (p *PodFacts) anyUninstalledTransientPodsNotRunning() (bool, types.Namespac
 	return false, types.NamespacedName{}
 }
 
-// getHostList will return a host list from the given pods
-func getHostList(podList []*PodFact) []string {
+// getHostList will return a host and podName list from the given pods
+func getHostAndPodNameList(podList []*PodFact) ([]string, []types.NamespacedName) {
 	hostList := make([]string, 0, len(podList))
+	podNames := make([]types.NamespacedName, 0, len(podList))
 	for _, pod := range podList {
 		hostList = append(hostList, pod.podIP)
+		podName := types.NamespacedName{Name: pod.name.Name}
```

Comment:
This ends up setting the pod name without a namespace. This is the reason for the error you were seeing. If you look at the output from the operator log you will see that the pod name its about to exec doesn't have the namespace:
```
kuttl-test-sound-humpback verticadb-operator-controller-manager-5d97bb85c8-hwqg9 manager 2023-09-21T01:22:14.200Z   INFO    controllers.VerticaDB   ExecInPod stream    {"verticadb": "kuttl-test-sound-humpback/v-http-custom-certs", "pod": "/v-http-custom-certs-main-0", "err": "pods \"v-http-custom-certs-main-0\" is forbidden: User \"system:serviceaccount:kuttl-test-sound-humpback:verticadb-operator-controller-manager\" cannot create resource \"pods/exec\" in API group \"\" at the cluster scope: RBAC: clusterrole.rbac.authorization.k8s.io \"verticadb-operator-metrics-reader\" not found", "stdout": "", "stderr": ""}
```

The namespaced name is already in the podfacts. You don't need to create a new one here.
```suggestion
		podName := pod.name
```

Reply:
Actually you don't even need `podName`. Pass in `pod.name` from podfacts when you build out the `podNames` list.

Reply:
Oh, I see! Thank you for your suggestion. I should notice it. Will fix it as your suggestion